homeassistant/components/vizio/config_flow.py

```diff
@@ -98,9 +98,9 @@ def _get_pairing_schema(input_dict: dict[str, Any] | None = None) -> vol.Schema:
 
 def _host_is_same(host1: str, host2: str) -> bool:
     """Check if host1 and host2 are the same."""
-    host1 = host1.split(":")[0]
+    host1 = host1.partition(":")[0]
```

Comment:
I am not sure how this comparison worked, since host1 and host2 can be IPv6 addresses and separating at the colon will just yield something like `00`.

This needs further investigation.  Meanwhile, this can be used to switch on the case that the input values are NOT IP addresses:

```
>>> ip_address("roxanne.dragonfear")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/ipaddress.py", line 54, in ip_address
    raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
ValueError: 'roxanne.dragonfear' does not appear to be an IPv4 or IPv6 address
```

Reply:
My guess is that Vizio equipment doesn't do IPv6, so that's a non-issue. 😁

Reply:
Let's postulate that.

Will we remember this issue in the function down the road 3 or 4 years from now, when Vizio TVs begin supporting IPv6?